---
title: "2024_1_22_DEA_Tumor_TumorAdj"
output: html_document
date: "2024-01-22"
---

**Purpose:**

* Identify HBV/HCV tumor/tumor-adjacent differentially expressed genes with only library type as a covariate

**Methods:**

* Limma voom pipeline on all tumor-tumor-adjcent comparison

* Only covariate is library type

**Results:**

## Load libraries

```{r, message=FALSE, warning=FALSE}
library(edgeR)
library(DESeq2)
library(limma)
library(ggplot2)
library(dplyr)
```

## Read in data

```{r}
metadata <- read.table("~/3.0 Hasting Research/Liver Cancer Analysis/Japan_HCC_metadata_QC_filtered.csv", row.names=1,header=TRUE, sep=",")
tumorAdjacentExp <- read.csv("~/3.0 Hasting Research/Liver Cancer Analysis/Japan_HCC_Salmon_Expression_QC_filtered.csv", row.names = 1, header=TRUE)
colnames(tumorAdjacentExp) <- gsub("\\.", "-", colnames(tumorAdjacentExp))
identical(metadata$sampleid, colnames(tumorAdjacentExp))

table(metadata$gender_tissue_viral)
```

## Relabeling genes
```{r}
# Importing gene annotations
genes <- read.table("~/3.0 Hasting Research/Liver Cancer Analysis/gencodeTranscripts.txt", header=TRUE, sep="\t")
genes <- data.frame(genes)
tumorAdjacentExp <- tumorAdjacentExp[rownames(tumorAdjacentExp) %in% genes$GENEID ,]
genes <- genes[match(rownames(tumorAdjacentExp), genes$GENEID),]
identical(rownames(tumorAdjacentExp), genes$GENEID)
# Calculating gene length, this is needed for calculating the FPKM values
genes$length <- with(genes, end - start)
```

## Create DEG list object
```{r}
dge <- DGEList(counts=tumorAdjacentExp, genes=genes)

colnames(dge) <- colnames(tumorAdjacentExp)
dge$samples$ID <- metadata$ID
dge$samples$tumor <- factor(metadata$tumor)
dge$samples$library_type <- metadata$libtype
dge$samples$viral <- metadata$Virus_infection
dge$samples$tissue_viral <- paste(metadata$Virus_infection, metadata$tumor, sep="_")
dge$samples$gender_tussue <- metadata$gender_tissue
dge$samples$gender_tissue_viral <- metadata$gender_tissue_viral
```

## Filtering by FPKM expression level
```{r}
# Keeping genes that have a mean FPKM of at least 0.5 in at least one of the
# groups under investigation and at least 6 reads in at least 10 samples
fpkm <- rpkm(dge, gene.length=dge$genes$length)

M_1_HBV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="M_1_HBV")],1,mean,na.rm=TRUE)
M_0_HBV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="M_0_HBV")],1,mean,na.rm=TRUE)
M_1_HCV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="M_1_HCV")],1,mean,na.rm=TRUE)
M_0_HCV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="M_0_HCV")],1,mean,na.rm=TRUE)

F_1_HBV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="F_1_HBV")],1,mean,na.rm=TRUE)
F_0_HBV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="F_0_HBV")],1,mean,na.rm=TRUE)
F_1_HCV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="F_1_HCV")],1,mean,na.rm=TRUE)
F_0_HCV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="F_0_HCV")],1,mean,na.rm=TRUE)

keep <- (M_1_HBV_mean_fpkm > 0.5 | M_0_HBV_mean_fpkm > 0.5 | 
           M_1_HCV_mean_fpkm > 0.5 | M_0_HCV_mean_fpkm > 0.5 |
           F_1_HBV_mean_fpkm > 0.5 | F_0_HBV_mean_fpkm > 0.5 |
           F_1_HCV_mean_fpkm > 0.5 | F_0_HCV_mean_fpkm > 0.5 )

dge <- dge[keep,,keep.lib.sizes=FALSE]
dge <- calcNormFactors(dge, method="TMM")
keep <- rowSums(dge$counts > 6) >= 10
dge <- dge[keep,,keep.lib.size=FALSE]
dge <- calcNormFactors(dge, method="TMM")

# N of genes retained after filtering
dim(dge$genes)
```

## Create design matrix
```{r}
design <- model.matrix(~0+dge$samples$tissue_viral+dge$samples$library_type)
colnames(design) <- gsub("dge\\$samples\\$tissue_viral", "", colnames(design))
colnames(design) <- gsub("dge\\$samples\\$library_typeunstranded", "library_type", colnames(design))
head(design)
```

## Test MAPK7 expression pre-voom
```{r}
test_expression_plot <- dge$counts
test_expression_plot <- as.data.frame(test_expression_plot)
MAPK7 <- test_expression_plot[which(row.names(test_expression_plot)=="ENSG00000166484.19"),]
MAPK7 <- t(MAPK7)
identical(row.names(MAPK7), metadata$sampleid)
MAPK7 <- as.data.frame(MAPK7)
MAPK7$group <- paste(metadata$Virus_infection, metadata$tumor, sep="_")
ggplot(MAPK7, aes(x=group, y=as.numeric(ENSG00000166484.19))) + geom_boxplot(outlier.shape = NA) + geom_jitter(width=0.05) + theme_minimal()

ggplot(MAPK7, aes(x=group, y=log(as.numeric(ENSG00000166484.19),10))) + geom_boxplot(outlier.shape = NA) + geom_jitter(width=0.05) + theme_minimal()
```

## Voom analysis
```{r}
v <- voomWithQualityWeights(dge, design, plot=TRUE)
```

## Test MAPK7 expression post-voom
```{r}
test_expression_plot <- v$E
test_expression_plot <- as.data.frame(test_expression_plot)
MAPK7 <- test_expression_plot[which(row.names(test_expression_plot)=="ENSG00000166484.19"),]
MAPK7 <- t(MAPK7)
identical(row.names(MAPK7), metadata$sampleid)
MAPK7 <- as.data.frame(MAPK7)
MAPK7$group <- paste(metadata$Virus_infection, metadata$tumor, sep="_")
ggplot(MAPK7, aes(x=group, y=as.numeric(ENSG00000166484.19))) + geom_boxplot(outlier.shape = NA) + geom_jitter(width=0.05) + theme_minimal()
```

## Test expression pre- versus post- voom

```{r}
test_expression_plot_pre_voom <- dge$counts
test_expression_plot_pre_voom <- as.data.frame(test_expression_plot_pre_voom)

MAPK7 <- test_expression_plot[which(row.names(test_expression_plot)=="ENSG00000166484.19"),]
MAPK7 <- t(MAPK7)
identical(row.names(MAPK7), metadata$sampleid)
MAPK7 <- as.data.frame(MAPK7)

MAPK7_pre_voom <- test_expression_plot_pre_voom[which(row.names(test_expression_plot_pre_voom)=="ENSG00000166484.19"),]
MAPK7_pre_voom <- t(MAPK7_pre_voom)
identical(row.names(MAPK7_pre_voom), metadata$sampleid)
MAPK7_pre_voom <- as.data.frame(MAPK7_pre_voom)

MAPK7$group <- paste(metadata$Virus_infection, metadata$tumor, sep="_")

MAPK_all <- cbind(MAPK7, MAPK7_pre_voom)
colnames(MAPK_all) <- c("Voom", "Group", "Pre_voom")
ggplot(MAPK_all, aes(x=Pre_voom, y=Voom, color=Group))+geom_point()+ theme_minimal() +geom_smooth(method="lm")
```

## Voom analysis
```{r}
v_mod <- voom(dge, plot=TRUE)
```

## Test MAPK7 expression post-voom without quality weights
```{r}
test_expression_plot <- v_mod$E
test_expression_plot <- as.data.frame(test_expression_plot)
MAPK7 <- test_expression_plot[which(row.names(test_expression_plot)=="ENSG00000166484.19"),]
MAPK7 <- t(MAPK7)
identical(row.names(MAPK7), metadata$sampleid)
MAPK7 <- as.data.frame(MAPK7)
MAPK7$group <- paste(metadata$Virus_infection, metadata$tumor, sep="_")
ggplot(MAPK7, aes(x=group, y=as.numeric(ENSG00000166484.19))) + geom_boxplot(outlier.shape = NA) + geom_jitter(width=0.05) + theme_minimal()
```

## MDS plots separated by sex and tissue type
```{r}
vCorrectLibtype <- removeBatchEffect(v, batch=v$targets$library_type)

identical(colnames(v), colnames(vCorrectLibtype))
vCorrectLibtype_T_F <- vCorrectLibtype[,which(v$targets$gender_tussue=="F_1")]
metadata_T_F <- metadata[match(colnames(vCorrectLibtype_T_F), metadata$sampleid),]
identical(colnames(vCorrectLibtype_T_F), metadata_T_F$sampleid)

mds <- plotMDS(vCorrectLibtype_T_F, top = 2000, dim.plot = c(1,2), plot=TRUE, cex=2, pch=1,
               col=ifelse(metadata_T_F$Virus_infection %in% c("HBV"), "darkgreen",
                          ifelse(metadata_T_F$Virus_infection %in% c("HCV"), "firebrick", "pink")),
               gene.selection = "common")#, xlim=c(1.5,2.5), ylim=c(-1.5,-1.0)) #, gene.selection = "pairwise",

pdf("~/3.0 Hasting Research/Liver Cancer Analysis/Figures/MDS_2000genes_MDS_tumor_female.pdf")
mds <- plotMDS(vCorrectLibtype_T_F, top = 2000, dim.plot = c(1,2), plot=TRUE, cex=2, pch=1,
               col=ifelse(metadata_T_F$Virus_infection %in% c("HBV"), "darkgreen",
                          ifelse(metadata_T_F$Virus_infection %in% c("HCV"), "firebrick", "pink")),
               gene.selection = "common")#, xlim=c(1.5,2.5), ylim=c(-1.5,-1.0)) #, gene.selection = "pairwise",
dev.off()

vCorrectLibtype_T_M <- vCorrectLibtype[,which(v$targets$gender_tussue=="M_1")]
metadata_T_M <- metadata[match(colnames(vCorrectLibtype_T_M), metadata$sampleid),]
identical(colnames(vCorrectLibtype_T_M), metadata_T_M$sampleid)

mds <- plotMDS(vCorrectLibtype_T_M, top = 2000, dim.plot = c(1,2), plot=TRUE, cex=2, pch=1,
               col=ifelse(metadata_T_M$Virus_infection %in% c("HBV"), "darkgreen",
                          ifelse(metadata_T_M$Virus_infection %in% c("HCV"), "firebrick", "pink")),
               gene.selection = "common")#, xlim=c(1.5,2.5), ylim=c(-1.5,-1.0)) #, gene.selection = "pairwise",

pdf("~/3.0 Hasting Research/Liver Cancer Analysis/Figures/MDS_2000genes_MDS_tumor_male.pdf")
mds <- plotMDS(vCorrectLibtype_T_M, top = 2000, dim.plot = c(1,2), plot=TRUE, cex=2, pch=1,
               col=ifelse(metadata_T_M$Virus_infection %in% c("HBV"), "darkgreen",
                          ifelse(metadata_T_M$Virus_infection %in% c("HCV"), "firebrick", "pink")),
               gene.selection = "common")#, xlim=c(1.5,2.5), ylim=c(-1.5,-1.0)) #, gene.selection = "pairwise",
dev.off()

vCorrectLibtype_A_F <- vCorrectLibtype[,which(v$targets$gender_tussue=="F_0")]
metadata_A_F <- metadata[match(colnames(vCorrectLibtype_A_F), metadata$sampleid),]
identical(colnames(vCorrectLibtype_A_F), metadata_A_F$sampleid)

mds <- plotMDS(vCorrectLibtype_A_F, top = 2000, dim.plot = c(1,2), plot=TRUE, cex=2, pch=1,
               col=ifelse(metadata_A_F$Virus_infection %in% c("HBV"), "darkgreen",
                          ifelse(metadata_A_F$Virus_infection %in% c("HCV"), "firebrick", "pink")),
               gene.selection = "common")#, xlim=c(1.5,2.5), ylim=c(-1.5,-1.0)) #, gene.selection = "pairwise",

pdf("~/3.0 Hasting Research/Liver Cancer Analysis/Figures/MDS_2000genes_MDS_adjacent_female.pdf")
mds <- plotMDS(vCorrectLibtype_A_F, top = 2000, dim.plot = c(1,2), plot=TRUE, cex=2, pch=1,
               col=ifelse(metadata_A_F$Virus_infection %in% c("HBV"), "darkgreen",
                          ifelse(metadata_A_F$Virus_infection %in% c("HCV"), "firebrick", "pink")),
               gene.selection = "common")#, xlim=c(1.5,2.5), ylim=c(-1.5,-1.0)) #, gene.selection = "pairwise",
dev.off()

vCorrectLibtype_A_M <- vCorrectLibtype[,which(v$targets$gender_tussue=="M_0")]
metadata_A_M <- metadata[match(colnames(vCorrectLibtype_A_M), metadata$sampleid),]
identical(colnames(vCorrectLibtype_A_M), metadata_A_M$sampleid)

mds <- plotMDS(vCorrectLibtype_A_M, top = 2000, dim.plot = c(1,2), plot=TRUE, cex=2, pch=1,
               col=ifelse(metadata_A_M$Virus_infection %in% c("HBV"), "darkgreen",
                          ifelse(metadata_A_M$Virus_infection %in% c("HCV"), "firebrick", "pink")),
               gene.selection = "common")#, xlim=c(1.5,2.5), ylim=c(-1.5,-1.0)) #, gene.selection = "pairwise",

pdf("~/3.0 Hasting Research/Liver Cancer Analysis/Figures/MDS_2000genes_MDS_adjacent_male.pdf")
mds <- plotMDS(vCorrectLibtype_A_M, top = 2000, dim.plot = c(1,2), plot=TRUE, cex=2, pch=1,
               col=ifelse(metadata_A_M$Virus_infection %in% c("HBV"), "darkgreen",
                          ifelse(metadata_A_M$Virus_infection %in% c("HCV"), "firebrick", "pink")),
               gene.selection = "common")#, xlim=c(1.5,2.5), ylim=c(-1.5,-1.0)) #, gene.selection = "pairwise",
dev.off()

mds <- plotMDS(vCorrectLibtype_A_M, top = 2000, dim.plot = c(1,2), plot=TRUE, cex=2, pch=1,
               col=ifelse(metadata_A_M$libtype %in% c("stranded"), "darkgreen",
                          ifelse(metadata_A_M$libtype %in% c("unstranded"), "firebrick", "pink")),
               gene.selection = "common")#, xlim=c(1.5,2.5), ylim=c(-1.5,-1.0)) #, gene.selection = "pairwise",

mds <- plotMDS(vCorrectLibtype_A_M, top = 2000, dim.plot = c(1,2), plot=TRUE, cex=2, pch=1,
               col=ifelse(metadata_A_M$Alcohol_intake %in% c(0), "darkgreen",
                          ifelse(metadata_A_M$Alcohol_intake %in% c(1,2,3), "firebrick", "pink")),
               gene.selection = "common")#, xlim=c(1.5,2.5), ylim=c(-1.5,-1.0)) #, gene.selection = "pairwise",

mds <- plotMDS(vCorrectLibtype_A_M, top = 2000, dim.plot = c(1,2), plot=TRUE, cex=2, pch=1,
               col=ifelse(metadata_A_M$Smoking %in% c(0), "darkgreen",
                          ifelse(metadata_A_M$Smoking %in% c(1,2,3), "firebrick", "pink")),
               gene.selection = "common")#, xlim=c(1.5,2.5), ylim=c(-1.5,-1.0)) #, gene.selection = "pairwise",

mds <- plotMDS(vCorrectLibtype_A_M, top = 2000, dim.plot = c(1,2), plot=TRUE, cex=2, pch=1,
               col=ifelse(metadata_A_M$Gender %in% c("M"), "darkgreen",
                          ifelse(metadata_A_M$Gender %in% c("F"), "firebrick", "pink")),
               gene.selection = "common")#, xlim=c(1.5,2.5), ylim=c(-1.5,-1.0)) #, gene.selection = "pairwise",
```

# DEA
```{r}
# Block design for individual. This is used in tumor-normal comparisons with
# paired samples.
corfit <- duplicateCorrelation(v, design, block = v$targets$ID)
# This should give a positive correlation value. It represents the
# correlation between measurements made on the same person.
corfit$consensus

# Fitting the linear model with limma.
# If using paired samples, the within-patient correlation and a block design
# for patient is used to account for pairwise samples
fit <- lmFit(v, design, block = v$targets$ID, correlation = corfit$consensus)

# Contrast design for differential expression
# Defining pairwise comparisons
contrasts <- makeContrasts(HBV_tumor_adjacent = HBV_1 - HBV_0,
                           HCV_tumor_ajcanet = HCV_1 - HCV_0,
                           levels=colnames(design))
head(contrasts)

# Running contrast analysis
vfit <- contrasts.fit(fit, contrasts = contrasts)

# Computing differential expression based on the empirical Bayes moderation of
# the standard errors towards a common value. Robust = should the estimation of
# the empirical Bayes prior parameters be robustified against outlier sample
# variances?
veBayesFit <- eBayes(vfit, robust=TRUE)
# Look at N of DEGs with adj. p <0.01 and log2FC>2
summary(decideTests(veBayesFit, adjust.method = "BH", p.value = 0.05, lfc = 2))

plotSA(veBayesFit, main = "Final model: Mean-variance trend")

vTopTable_HBV <- topTable(veBayesFit, n=Inf, p.value=1, lfc=0, coef = 1)
write.csv(vTopTable_HBV, "~/3.0 Hasting Research/Liver Cancer Analysis/DEGs/HBV_tumor_tumor_adj_p1_lfc0.csv")

DEGs_HBV <- topTable(veBayesFit, n=Inf, p.value=0.05, lfc=2, coef=1)
DEGs_HBV_print <- data.frame(DEGs_HBV$GENEID, DEGs_HBV$gene_name, DEGs_HBV$adj.P.Val, DEGs_HBV$logFC)
write.csv(DEGs_HBV_print, "~/3.0 Hasting Research/Liver Cancer Analysis/DEGs/HBV_tumor_tumor_adj_p0.05_lfc2.csv")

vTopTable_HCV <- topTable(veBayesFit, n=Inf, p.value=1, lfc=0, coef = 2)
write.csv(vTopTable_HCV, "~/3.0 Hasting Research/Liver Cancer Analysis/DEGs/HCV_tumor_tumor_adj_p1_lfc0.csv")

DEGs_HCV <- topTable(veBayesFit, n=Inf, p.value=0.05, lfc=2, coef=2)
DEGs_HCV_print <- data.frame(DEGs_HCV$GENEID, DEGs_HCV$gene_name, DEGs_HCV$adj.P.Val, DEGs_HCV$logFC)
write.csv(DEGs_HCV_print, "~/3.0 Hasting Research/Liver Cancer Analysis/DEGs/HCV_tumor_tumor_adj_p0.05_lfc2.csv")
```

# DEA
```{r}
# Block design for individual. This is used in tumor-normal comparisons with
# paired samples.
corfit <- duplicateCorrelation(v, design, block = v$targets$ID)
# This should give a positive correlation value. It represents the
# correlation between measurements made on the same person.
corfit$consensus

# Fitting the linear model with limma.
# If using paired samples, the within-patient correlation and a block design
# for patient is used to account for pairwise samples
fit <- lmFit(v, design, block = v$targets$ID, correlation = corfit$consensus)

# Contrast design for differential expression
# Defining pairwise comparisons
contrasts <- makeContrasts(HBV_HCV = HBV_1 - HCV_1,
                           levels=colnames(design))
head(contrasts)

# Running contrast analysis
vfit <- contrasts.fit(fit, contrasts = contrasts)

# Computing differential expression based on the empirical Bayes moderation of
# the standard errors towards a common value. Robust = should the estimation of
# the empirical Bayes prior parameters be robustified against outlier sample
# variances?
veBayesFit <- eBayes(vfit, robust=TRUE)
# Look at N of DEGs with adj. p <0.01 and log2FC>2
summary(decideTests(veBayesFit, adjust.method = "BH", p.value = 0.05))

plotSA(veBayesFit, main = "Final model: Mean-variance trend")

vTopTable <- topTable(veBayesFit, n=Inf, p.value=0.05, lfc=0, coef = 1)

vCorrectLibtype_abbr <- vCorrectLibtype[match(vTopTable$GENEID, rownames(vCorrectLibtype)),]

identical(rownames(vCorrectLibtype_abbr), vTopTable$GENEID)

mds <- plotMDS(vCorrectLibtype_abbr, top = 1000, dim.plot = c(1,2), plot=TRUE, cex=2, pch=1,
               col=ifelse(metadata$Virus_infection %in% c("HBV") & metadata$tumor==1, "darkgreen",
                          ifelse(metadata$Virus_infection %in% c("HBV") & metadata$tumor==0, "lightgreen",
                          ifelse(metadata$Virus_infection %in% c("HCV") & metadata$tumor==1, "lightblue",
                                 ifelse(metadata$Virus_infection %in% c("HCV") & metadata$tumor==0, "darkblue","pink")))),
               gene.selection = "common")#, xlim=c(1.5,2.5), ylim=c(-1.5,-1.0)) #, gene.selection = "pairwise",

pdf("~/3.0 Hasting Research/Liver Cancer Analysis/Figures/MDS_HBV_HCV_DEGs.pdf")
mds <- plotMDS(vCorrectLibtype_abbr, top = 1000, dim.plot = c(1,2), plot=TRUE, cex=2, pch=1,
               col=ifelse(metadata$Virus_infection %in% c("HBV") & metadata$tumor==1, "darkgreen",
                          ifelse(metadata$Virus_infection %in% c("HBV") & metadata$tumor==0, "lightgreen",
                          ifelse(metadata$Virus_infection %in% c("HCV") & metadata$tumor==1, "lightblue",
                                 ifelse(metadata$Virus_infection %in% c("HCV") & metadata$tumor==0, "darkblue","pink")))),
               gene.selection = "common")#, xlim=c(1.5,2.5), ylim=c(-1.5,-1.0)) #, gene.selection = "pairwise",
dev.off()


mds <- plotMDS(vCorrectLibtype_abbr, top = 1000, dim.plot = c(1,2), plot=TRUE, cex=2,
               col=ifelse(metadata$Virus_infection %in% c("HBV") & metadata$tumor==1, "darkgreen",
                          ifelse(metadata$Virus_infection %in% c("HBV") & metadata$tumor==0, "lightgreen",
                          ifelse(metadata$Virus_infection %in% c("HCV") & metadata$tumor==1, "lightblue",
                                 ifelse(metadata$Virus_infection %in% c("HCV") & metadata$tumor==0, "darkblue","pink")))),
               pch= ifelse(metadata$Gender %in% c("M"), 2,1),
               gene.selection = "common")#, xlim=c(1.5,2.5), ylim=c(-1.5,-1.0)) #, gene.selection = "pairwise",
```

**Interpretation:**

* Model fit looks good before and after voom adjustment

* Positive correlation for samples from the same patient

* DEG lists saved


